src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unusedcatchparametershouldbeunnamed/InputUnusedCatchParameterShouldBeUnnamed.java

```diff
@@ -0,0 +1,118 @@
+/*
+UnusedCatchParameterShouldBeUnnamed
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.coding.unusedcatchparametershouldbeunnamed;
+
+public class InputUnusedCatchParameterShouldBeUnnamed {
+    private E e;
+    Exception exception;
+
+    void test(Object o) {
+        try {
+            int x = 1/ 0;
+        }
+        catch (Exception e) { // violation
+            System.out.println("infinity");
+        }
+
+        try {
+            int x = 1/ 0;
+        }
+        catch (Exception e) {  // violation
+
+        }
+
+        try {
+            int x = 1/ 0;
+        }
+        catch (Exception e) {  // violation
+            this.e.printStackTrace();
+        }
+
+        try {
+            int x = 1/ 0;
+        }
+        catch (Exception e) {
+            System.out.println(e);
+        }
+
+        try {
+            int x = 1/ 0;
+        }
+        catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+
+        try {
+            int x = 1/ 0;
+        }
+        catch (Exception e) {  // violation
+           e = new Exception();
+        }
+        try {
+            int x = 1/ 0;
+        }
+        catch (Exception e) {
+            exception = e;
+        }
+
+        try {
+            int x = 1/ 0;
+        }
+        catch (Exception A) {  // violation
+           A a;
+        }
+
+        try {
+            int x = 1/ 0;
+        }
+        catch (Exception A) {  // violation
+           Object a = new A();
```

Comment:
Is there an example of an anonymous class where it has a method with its own try/catch of the same name and it is used?

Reply:
I am not sure but do you mean something like this?

https://github.com/mahfouz72/checkstyle/blob/b0371828cfebfe14b569d80f54f44af9ad99c8af/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unusedcatchparametershouldbeunnamed/InputUnusedCatchParameterShouldBeUnnamedNested.java#L102-L111

Reply:
Yes, but put that block inside the catch and have both catch be the same variable name.

I am basically looking for a way to confuse your check with flagging something it shouldn't or vice versa.

Example (lazy indentation):
````
            try {
                int y = 1 / 0;
            } catch (Exception ex) { // should be violation

    // nested inside anonymous class
    Runnable runnable = new Runnable() {
        @Override
        public void run() {
            try {
                int x = 1 / 0;
            } catch (Exception ex) { // should be ok
                System.out.println(ex.getMessage());
            }
        }
    };

            }
````

Reply:
done

Reply:
https://github.com/mahfouz72/checkstyle/blob/be69edc8c69a195ef25c897afc6c6d50b3c79f90/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unusedcatchparametershouldbeunnamed/InputUnusedCatchParameterShouldBeUnnamedNested.java#L101-L114